diff --git a/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppBusinessShare.java b/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppBusinessShare.java
index 637b192..0613444 100644
--- a/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppBusinessShare.java
+++ b/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppBusinessShare.java
@@ -35,5 +35,6 @@ public class WhatsAppBusinessShare extends SingleShareIntent {
                 // leave room for the conversation to be created
                 Thread.sleep(START_ACTIVITY_TIME_GAP_MS);   
+                return;
             } catch (Exception ex) {
                 ex.printStackTrace();
             }
diff --git a/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppShare.java b/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppShare.java
index 742e8f9..bcf3303 100644
--- a/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppShare.java
+++ b/node_modules/react-native-share/android/src/main/java/cl/json/social/WhatsAppShare.java
@@ -35,5 +35,6 @@ public class WhatsAppShare extends SingleShareIntent {
                 // leave room for the conversation to be created
                 Thread.sleep(START_ACTIVITY_TIME_GAP_MS);   
+                return;
             } catch (Exception ex) {
                 ex.printStackTrace();
             }
